deployment/cdk/opensearch-service-migration/lib/historical-capture-stack.ts

```diff
@@ -1,62 +1,97 @@
+            logging: LogDrivers.awsLogs({ streamPrefix: 'fetch-migration-lg', logRetention: 30 })
+        });
+
+        // Create DP pipeline config from template file
+        let dpPipelineData: string = readFileSync(props.dpPipelineTemplatePath, 'utf8');
+        dpPipelineData = dpPipelineData.replace("<SOURCE_CLUSTER_HOST>", props.sourceEndpoint);
+        dpPipelineData = dpPipelineData.replace("<TARGET_CLUSTER_HOST>", props.targetEndpoint);
+        // Base64 encode
+        let encodedPipeline = Buffer.from(dpPipelineData).toString("base64");
+
+        // Create secret using Secrets Manager
+        const dpPipelineConfigSecret = new SMSecret(this, "dpPipelineConfigSecret", {
+            secretName: `${historicalCaptureFargateTask.family}-${historicalCaptureContainer.containerName}-pipelineConfig`,
+            secretStringValue: SecretValue.unsafePlainText(encodedPipeline)
         });
+        // Add secret to container
+        historicalCaptureContainer.addSecret("INLINE_PIPELINE", 
+            ECSSecret.fromSecretsManager(dpPipelineConfigSecret)
+        );
 
-        // Create Fargate Service
-        const historicalCaptureFargateService = new FargateService(this, "historicalCaptureFargateService", {
-            cluster: ecsCluster,
-            taskDefinition: historicalCaptureFargateTask,
-            desiredCount: 1
+        // The Migration Domain SG disallows all outgoing, but this prevents the ECS run task
+        // command from pulling the Docker image from ECR. Thus, create an SG to allow this.
+        // TODO - Limit outbound to only ECR endpoints somehow
+        const fetchMigrationSecurityGroup = new SecurityGroup(this, 'fetchMigrationECRAccessSG', {
+            vpc: ecsCluster.vpc
         });
 
+        // CFN outputs that will be consumed by Copilot
+        new CfnOutput(this, 'FMTaskDefArn', {
+            exportName: 'FetchMigrationTaskDefinitionArn',
+            value: historicalCaptureFargateTask.taskDefinitionArn,
+            description: 'ARN of the ECS Task Definition for Fetch Migration',
+        });
+        new CfnOutput(this, 'FMTaskRoleArn', {
+            exportName: 'FetchMigrationTaskRoleArn',
+            value: historicalCaptureFargateTask.taskRole.roleArn,
+            description: 'ARN of the Task Role for Fetch Migration',
+        });
+        new CfnOutput(this, 'FMExecRoleArn', {
+            exportName: 'FetchMigrationExecutionRoleArn',
+            value: historicalCaptureFargateTask.obtainExecutionRole().roleArn,
+            description: 'ARN of the Execution Role for Fetch Migration',
+        });
+        // Documentation - https://docs.aws.amazon.com/cli/latest/reference/ecs/run-task.html#options
+        let networkConfigJson = {
+            "awsvpcConfiguration": {
+                "subnets": ecsCluster.vpc.privateSubnets.map(_ => _.subnetId),
+                "securityGroups": [process.env.MIGRATION_DOMAIN_SG_ID, fetchMigrationSecurityGroup.securityGroupId]
```

Comment:
Same as before, surprised you can access the MIGRATION_DOMAIN_SG_ID in this way

Reply:
As noted in the [other comment](https://github.com/opensearch-project/opensearch-migrations/pull/346#discussion_r1355753633), this accesses the exported environment variable. Ideally, we would use CFN outputs for this.